File: Data Structures & Algorithms/binary-search/submission-14.cs
public class Solution {
public int Search(int[] nums, int target)
{
    var l = 0;
    var r = nums.Length - 1;
    var mid = r/2;
    while (l <= r)
    {
        if (target == nums[mid])
            return mid;
        else if (target > nums[mid])
            l = mid + 1;
        else 
            r = mid - 1;

        mid = (r + l) / 2;
    };

    return -1;
}
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-14.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l, r = 0, len(nums)-1
        count_rotated = 0
        while(l < r):
            isRotated = nums[l] > nums[r] 
            if isRotated:
                count_rotated += 1
                l += 1 
            else:
                break


        return nums[count_rotated]